### PR TITLE
schemas loop reference

### PR DESCRIPTION
**Pull request by LuisTejedaS**
## Description

While trying to sort the schemas by dependency there is a case when the schemas have imports that lead to an infinite loop:

```
        <xsd:schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.xsd2jsonschema.org/example">
            <xsd:import namespace="http://www.xsd2jsonschema.org/example2"/>
            <xsd:import namespace="http://www.xsd2jsonschema.org/example3"/>
        </xsd:schema>
        <xsd:schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.xsd2jsonschema.org/example2">
            <xsd:import namespace="http://www.xsd2jsonschema.org/example"/>
            <xsd:import namespace="http://www.xsd2jsonschema.org/example3"/>
        </xsd:schema>
        <xsd:schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.xsd2jsonschema.org/example3">
            <xsd:import namespace="http://www.xsd2jsonschema.org/example"/>
            <xsd:import namespace="http://www.xsd2jsonschema.org/example2"/>
        </xsd:schema>
```

We were trying to avoid the Maximum stack limit reached comparing the imported schema with the parent so this worked for scenarios like this:

```
        <xsd:schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.xsd2jsonschema.org/example">
            <xsd:import namespace="http://www.xsd2jsonschema.org/example2"/>
        </xsd:schema>
        <xsd:schema elementFormDefault="qualified" xmlns="http://www.w3.org/2001/XMLSchema"  
              targetNamespace="http://www.xsd2jsonschema.org/example2">
                <xsd:import namespace="http://www.xsd2jsonschema.org/example"/>
       </xsd:schema>
```

but not for a deeper loop

##Solution
Now we are storing the processed schemas in a list, so if we are about to import a new schema and is already in the list we skip the recursive call

## How to test this ticket

- NPM run test

### CODE REVIEW

File: lib/utils/orderSchemasByDependency.js [hunk 1]
```diff
@@ -130,9 +130,10 @@ function processSchema(schemaAndInformation, independent, attributePlaceHolder,
       schemaAndInformation.dependencies.add(
         getXMLAttributeByName(importElement, attributePlaceHolder, 'namespace'));
       assignNamespacesToSchema(schemaAndInformation, namespacesFromWSDL, attributePlaceHolder);
-      if (toProcess && parentTargetNamespace !== importedName) {
+      if (toProcess && processedSchemas.find((schema) => { return importedName === schema; }) === undefined) {
```
Comment (erfemega) [inline lib/utils/orderSchemasByDependency.js:133]: Maybe we could take it: 'processedSchemas.find((schema) => { return importedName === schema; }) === undefined' out to a variable.

let isNotProcessedSchema = !processedSchemas.find((schema) => { return importedName === schema; })

Comment (LuisTejedaS) [reply lib/utils/orderSchemasByDependency.js]: Yes, i think creating a variable reflects better the intention

**Review by erfemega — APPROVED** [review]
Nice Job

**Review by VShingala — APPROVED** [review]
LGTM!

**Pull request merged**